CMakeLists.txt

```diff
@@ -406,7 +493,7 @@ add_option("WOLFSSL_AESGCM"
     "yes" "yes;no;table;small;word32;4bit")
 
 # leanpsk and leantls don't need gcm
-if(WOLFSSL_LEAN_PSK OR (WOLFSSL_LEAN_TLS AND NOT WOLFSSL_TLS13))
+if(BUILD_FIPS_RAND OR WOLFSSL_LEAN_PSK OR (WOLFSSL_LEAN_TLS AND NOT WOLFSSL_TLS13))
```

Comment:
Use `"${FIPS_VERSION}" STREQUAL "rand"` instead of BUILD_FIPS_RAND here. BUILD_FIPS_RAND isn't available here.